vcd/resource_vcd_cse_kubernetes_cluster.go

```diff
@@ -0,0 +1,796 @@
+package vcd
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"github.com/hashicorp/go-cty/cty"
+	semver "github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"sort"
+	"time"
+)
+
+func resourceVcdCseKubernetesCluster() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdCseKubernetesClusterCreate,
+		ReadContext:   resourceVcdCseKubernetesRead,
+		UpdateContext: resourceVcdCseKubernetesUpdate,
+		DeleteContext: resourceVcdCseKubernetesDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdCseKubernetesImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"cse_version": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"4.1.0", "4.1.1", "4.2.0"}, false),
```

Comment:
Any risk of overconstraining for something like `4.1.1a`?

Reply:
That's a good point. I can add `4.1.1a` if that avoids possible misunderstandings.

Internally, it is treated as `4.1.1`. I didn't add it as practically is the same release (equivalent in everything but the name).

@lvirbalas @dataclouder What do you think?

Reply:
If it is treated as `4.1.1` I guess we could just put in the docs and leave the code as it is. We won't be able to guess all possible patch releases IMO.

Reply:
Yeah, actually the difference between 4.1.1 and 4.1.1a is only the CSE Server OVA used during the installation.

Reply:
I have stated in the resource and data source documentations that it supports "4.1.1a", and for the `cse_version` argument 4.1.1 == 4.1.1a